### PR TITLE
#75 cypress new building operation

### LINKED ISSUE #75: Task: Create user acceptance test for "Add New Building Operation" page.
#61

### PR DESCRIPTION
**Pull request by IsabelleCzuboka**
This cypress test is about adding a new operation to a building. I created a function to delete an operation so that the test may run multiple times without fault.

closes #75

### CODE REVIEW

**Review by npiperni — APPROVED** [review]

**Review by amczuboka — APPROVED** [review]
Changes look good

**Review by Tharsipa — APPROVED** [review]

**Review by CarlaShawi — APPROVED** [review]

File: src/app/services/building.service.ts [hunk 1]
```diff
@@ -284,6 +285,55 @@ export class BuildingService {
     }
   }
 
+
+  async deleteOperationByName(buildingId: string, operationName: string): Promise<void> {
```
Comment (Holyvert) [inline src/app/services/building.service.ts:289]: where is the unit test

**Review by Holyvert — CHANGES_REQUESTED** [review]

**Review by Guillaume-Lachapelle — APPROVED** [review]
Same comment as Oliver. The rest looks good

**Review by Ziadsharkos — APPROVED** [review]

**Comment by amczuboka** [comment]
> I think it would be better to delete the operation by ID to ensure it is unique and we don't accidentally delete an operation with the same name.

I don't think this is an issue, because it's only being used in a cypress test. If we wanted to delete it by ID, we would have to delete every operation anyway since there's no way to get the specific operation. For example, in the createBuilding.cy test it deletes every building of the user even if it only created 1 building
```     
 const promiseDelete = user.PropertyIds.map(async (ID) => {
        await(win as any).buildingService.deleteBuilding(ID);
  });
```

**Comment by KarinaSandur** [comment]
> > I think it would be better to delete the operation by ID to ensure it is unique and we don't accidentally delete an operation with the same name.
> 
> I don't think this is an issue, because it's only being used in a cypress test. If we wanted to delete it by ID, we would have to delete every operation anyway since there's no way to get the specific operation. For example, in the createBuilding.cy test it deletes every building of the user even if it only created 1 building
> 
> ```
>  const promiseDelete = user.PropertyIds.map(async (ID) => {
>         await(win as any).buildingService.deleteBuilding(ID);
>   });
> ```

If that's the case, then I'll approve

**Review by KarinaSandur — APPROVED** [review]

**Review by Holyvert — APPROVED** [review]

**Pull request merged**